include/libtorrent/stack_allocator.hpp

```diff
@@ -62,29 +62,31 @@ namespace libtorrent { namespace aux
 		int copy_buffer(char const* buf, int size)
 		{
 			int ret = int(m_storage.size());
+			if (size < 1) return -1;
 			m_storage.resize(ret + size);
 			memcpy(&m_storage[ret], buf, size);
 			return ret;
 		}
 
 		int allocate(int bytes)
 		{
+			if (bytes < 1) return -1;
 			int ret = int(m_storage.size());
 			m_storage.resize(ret + bytes);
 			return ret;
 		}
 
 		char* ptr(int idx)
 		{
-			TORRENT_ASSERT(idx >= 0);
-			TORRENT_ASSERT(idx < int(m_storage.size()));
+			if(idx < 0) return NULL;
+			if(idx >= int(m_storage.size())) return NULL;
```

Comment:
I think this second test may still may sense to leave as an assert. I don't see a case where this would trigger without having a bug somewhere